include/gatekeeper_gk.h

```diff
@@ -99,17 +99,26 @@ struct neighbor_hash_table {
 	/* The tables that store the Ethernet headers. */
 	struct ether_cache *cache_tbl;
 
-	/* TODO Use a spin lock to edit the hash table. */
 	struct rte_hash *hash_table;
 };
 
 /* The gk forward information base (fib). */
 struct gk_fib {
 
+	/* Use a spin lock to edit the FIB table. */
+	rte_spinlock_t lock;
```

Comment:
(1) Since we only maintain one FIB table, all the GK blocks can ``read`` any FIB entry, moreover, the dynamic configuration and CPS blocks can ``write`` any FIB entry. So, we need to synchronize their behaviors.
In one extreme case, we may choose to just use one lock for the FIB table, once there is a write, all the readers need to wait until the write completes, which seems pretty expensive in terms of performance.
In another extreme case, like I did here, one lock per FIB, this way, the writers can only block the readers who want to access that FIB entry, all other readers can proceed without any block.
Possibly, there is a trade-off. We may maintain a table of locks, and find a way to map a FIB entry to one lock in the table?

(2) I will use the transaction memory locks.